<?xml version="1.0"?>
<!--
  This file is supposed to launch the drivers of the real robots instead of the simulation.
-->
<launch>

  <!-- Publish robot model -->
  <include file="$(find o2as_scene_description)/launch/o2as_base_scene_upload.launch" />
  <param name="use_real_robot" type="bool" value="true" />

  <!-- Required for the UR scripts to be sent -->
  <param name="use_real_robot" type="bool" value="true" />

  <!-- robot_ip: IP-address of the robot's socket-messaging server -->
  <!-- <arg name="robot_ip_a" default="10.66.171.51"/> -->
  <arg name="robot_ip_b" default="10.66.171.51"/>
  <!-- <arg name="robot_ip_c" default="192.168.0.41"/> -->
  <arg name="limited" default="false"/>
  <arg name="min_payload"  default="0.0"/>
  <arg name="max_payload"  default="3.0"/>
  <arg name="servoj_time" default="0.008" />
  <arg name="shutdown_on_disconnect" default="true" />
  <!-- require_activation defines when the service /ur_driver/robot_enable needs to be called. -->
  <arg name="require_activation" default="Never" /> <!-- Never, Always, OnStartup -->
    
  <!-- The max_velocity parameter is only used for debugging in the ur_driver. It's not related to actual velocity limits -->
  <arg name="max_velocity" default="10.0"/> <!-- [rad/s] -->

  <!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" /> -->

  <!-- Drivers for the robots -->
  <!-- <node name="a_bot_ur_driver" pkg="ur_modern_driver" type="ur_driver" output="screen" ns="a_bot_controller">
    <param name="prefix" type="str" value="a_bot_" />
    <param name="robot_ip_address" type="str" value="$(arg robot_ip_a)" />
    <param name="reverse_port" type="int" value="50003" /> 
    <param name="min_payload" type="double" value="$(arg min_payload)" />
    <param name="max_payload" type="double" value="$(arg max_payload)" />
    <param name="max_velocity" type="double" value="$(arg max_velocity)" />
    <param name="servoj_time" type="double" value="$(arg servoj_time)" />
    <param name="base_frame" type="str" value="a_bot_base" />
    <param name="tool_frame" type="str" value="a_bot_tool0_controller" />
    <param name="require_activation" type="str" value="$(arg require_activation)" />
    <param name="shutdown_on_disconnect" type="bool" value="$(arg shutdown_on_disconnect)"/>
    <remap from="joint_states" to="/joint_states"/>
  </node> -->
  
  <node name="b_bot_ur_driver" pkg="ur_modern_driver" type="ur_driver" output="screen" ns="b_bot_controller">
    <param name="prefix" type="str" value="b_bot_" />
    <param name="robot_ip_address" type="str" value="$(arg robot_ip_b)" />
    <param name="reverse_port" type="int" value="50002" /> 
    <param name="min_payload" type="double" value="$(arg min_payload)" />
    <param name="max_payload" type="double" value="$(arg max_payload)" />
    <param name="max_velocity" type="double" value="$(arg max_velocity)" />
    <param name="servoj_time" type="double" value="$(arg servoj_time)" />
    <param name="base_frame" type="str" value="b_bot_base" />
    <param name="tool_frame" type="str" value="b_bot_tool0_controller" />
    <!-- <param name="tool_frame" type="str" value="b_bot_dual_suction_gripper_pad_link" /> -->
    <param name="require_activation" type="str" value="$(arg require_activation)" />
    <param name="shutdown_on_disconnect" type="bool" value="$(arg shutdown_on_disconnect)"/>
    <remap from="joint_states" to="/joint_states"/>
  </node>
  
  <!-- <node name="c_bot_ur_driver" pkg="ur_modern_driver" type="ur_driver" output="screen" ns="c_bot_controller">
    <param name="prefix" type="str" value="c_bot_" />
    <param name="robot_ip_address" type="str" value="$(arg robot_ip_c)" />
    <param name="reverse_port" type="int" value="50001" /> 
    <param name="min_payload" type="double" value="$(arg min_payload)" />
    <param name="max_payload" type="double" value="$(arg max_payload)" />
    <param name="max_velocity" type="double" value="$(arg max_velocity)" />
    <param name="servoj_time" type="double" value="$(arg servoj_time)" />
    <param name="base_frame" type="str" value="c_bot_base" />
    <param name="tool_frame" type="str" value="c_bot_tool0_controller" />
    <param name="require_activation" type="str" value="$(arg require_activation)" />
    <param name="shutdown_on_disconnect" type="bool" value="$(arg shutdown_on_disconnect)"/>
    <remap from="joint_states" to="/joint_states"/>
  </node> -->
  
  <!-- Grippers -->
  <!-- <include file="$(find robotiq_control)/launch/ur_cmodel_action_controller.launch">
    <arg name="namespace" value="b_bot_gripper" />
    <arg name="gripper_prefix" value="b_bot_" />
    <arg name="topic" value="/b_bot_controller/ur_driver/URScript" />
  </include>

  <include file="$(find robotiq_control)/launch/cmodel_action_controller.launch">
    <arg name="ip" default="192.168.0.40" />
    <arg name="gripper_prefix" value="c_bot_" />
    <arg name="namespace" value="c_bot_gripper" />
  </include>

  <include file="$(find o2as_precision_gripper)/launch/precision_gripper.launch">
    <arg name="serial_port" default="/dev/ttyUSB2" />
  </include> -->

  <!-- <include file="$(find o2as_aist_config)/launch/o2as_moveit_planning_execution.launch"> -->
  <!--   <arg name="sim" value="false" /> -->
  <!--   <arg name="debug" value="false" /> -->
  <!--   <arg name="config" value="true" /> -->
  <!-- </include> -->

</launch>
